uaclient/tests/test_util.py

```diff
@@ -474,57 +474,216 @@ def test_get_platform_info_with_version(
-                "b": "bx1",
-                "c": "c1",
+                "affordances": {"some_affordance": [expected_value]}
             }
         }
-        util.apply_series_overrides(orig_access)
+
+        util.apply_contract_overrides(orig_access)
         assert orig_access == expected
 
     @mock.patch(
-        "uaclient.util.get_platform_info", return_value={"series": "xenial"}
+        "uaclient.util.get_platform_info", return_value={"series": "ubuntuX"}
+    )
+    @mock.patch(
+        "uaclient.clouds.identity.get_cloud_type", return_value="cloudX"
     )
-    def test_missing_keys_are_handled(self, _):
+    def test_different_overrides_applied_together(
+        self, _m_cloud_type, _m_platform_info
+    ):
+        """Apply different overrides from different matching selectors."""
         orig_access = {
             "entitlement": {
-                "series": {"xenial": {"directives": {"suites": ["xenial"]}}}
+                "affordances": {"some_affordance": ["original_affordance"]},
+                "directives": {"some_directive": ["original_directive"]},
+                "obligations": {"some_obligation": False},
+                "series": {
+                    "ubuntuX": {
+                        "affordances": {
+                            "new_affordance": ["new_affordance_value"]
+                        }
+                    }
+                },
+                "overrides": [
+                    {
+                        "selector": {"series": "ubuntuX"},
+                        "affordances": {
+                            "some_affordance": ["series_overriden"]
+                        },
+                    },
+                    {
+                        "selector": {"cloud": "cloudX"},
+                        "directives": {"some_directive": ["cloud_overriden"]},
+                    },
+                    {
+                        "selector": {"series": "ubuntuX", "cloud": "cloudX"},
+                        "obligations": {
+                            "new_obligation": True,
+                            "some_obligation": True,
+                        },
+                    },
+                ],
             }
         }
-        expected = {"entitlement": {"directives": {"suites": ["xenial"]}}}
 
-        util.apply_series_overrides(orig_access)
+        expected = {
+            "entitlement": {
+                "affordances": {
+                    "new_affordance": ["new_affordance_value"],
+                    "some_affordance": ["series_overriden"],
+                },
+                "directives": {"some_directive": ["cloud_overriden"]},
+                "obligations": {
+                    "new_obligation": True,
+                    "some_obligation": True,
+                },
+            }
+        }
 
-        assert expected == orig_access
+        util.apply_contract_overrides(orig_access)
+        assert orig_access == expected
```

Comment:
Should we add a test where we replace something with the selector ?

Reply:
This test covers the case where we add new things and replace some others (like the `some_affordance`). Sorry, maybe I missed your point here

Reply:
My mistake, I missed that the list will be completely replaced. Just ignore this